src/fields/jsgrid.field.control.js

```diff
@@ -91,16 +93,16 @@
         },
 
         filterTemplate: function() {
-            var $result = this._createSearchButton();
+            var $result = this.searchButton ? this._createSearchButton() : $("");
             return this.clearFilterButton ? $result.add(this._createClearFilterButton()) : $result;
         },
 
         insertTemplate: function() {
-            return this._createInsertButton();
+            return this.insertButton ? this._createInsertButton() : $("");
         },
 
         editTemplate: function() {
-            return this._createUpdateButton().add(this._createCancelEditButton());
+            return this.editButton ? this._createUpdateButton().add(this._createCancelEditButton()) : $("");
```

Comment:
Not sure, why do we need to hide update & cancel button for edit row? 
Initial idea was hiding edit button, but still being able to edit a row (e.g. with row click). And to complete editing or cancel changed we need these buttons.